### PR TITLE
Updated .gitignore file

### PR DESCRIPTION
**Pull request by PattyCakesxD**
I sort of just copy and pasted the following gitignores, other than that I have no clue what I'm doing:

- Node, MySQL, Docker, GraphQL, VisualStudio

### CODE REVIEW

**Comment by misslame** [comment]
Did you add every possible scenario file you can think of to being ignored? Like I saw c++ and .NET generated files there.

**Comment by misslame** [comment]
Also did you check if it does indeed ignore the files we care about. Lol I would check myself but im on my phone and it scrolls for days.

**Comment by PattyCakesxD** [comment]
Didn't know what qualifies as valid files to ignore unless it requires extensive research (I know nothing about MySQL, Node, and Docker) so I just copy and pasted the gitignores of the examples in the task. Is this really as easy as it's marked?

**Comment by PattyCakesxD** [comment]
Updated .gitignore again

A lot of files were irrelevant to THIS sprint, but I kept the ones that are generally important for each respective category (albeit I deemed some files as important even though not 100% sure on their purpose). If the .gitignore only needs to pertain exactly for what's required in this sprint, the custom .gitignore would be about 5 lines long. Let me know if I did anything wrong with the update please!

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,72 @@
...
 node_modules/
+jspm_packages/
+
+# TypeScript cache
+*.tsbuildinfo
+
+# Optional eslint cache
+.eslintcache
+
+# dotenv environment variable files
```
Comment (GuyWhoCode) [inline .gitignore:27]: This is an optional trim if the desired outcome is to reduce the amount of files in `.gitignore`. As far as I'm aware, `.env` should be the only file used for environment variables.  Local / production environments usually have their own instance of a `.env` file.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,72 @@
...
+.env
+.env.development.local
+.env.test.local
+.env.production.local
+.env.local
+
+# Stores VSCode versions used for testing VSCode extensions
+.vscode-test
+
+# ------- MySQL ignore -------
```
Comment (GuyWhoCode) [inline .gitignore]: I haven't encountered any of these files when writing MySQL for my issue. These files under the `MySQL ignore` can be removed.

**Review by GuyWhoCode — CHANGES_REQUESTED** [review]

**Review by misslame — APPROVED** [review]

**Review by GuyWhoCode — APPROVED** [review]

**Pull request merged**